src/app/labeling/qgsmaptoollabel.cpp

```diff
@@ -990,7 +1138,7 @@ bool QgsMapToolLabel::createAuxiliaryFields( LabelDetails &details, QgsDiagramIn
     {
       index = vlayer->fields().lookupField( prop.field() );
     }
-    else
+    else if ( prop.propertyType() != QgsProperty::ExpressionBasedProperty )
```

Comment:
```suggestion
    else
```
see above for reason